### ISSUE TITLE
[Bug Report]: embedding_bw returns wrong shape (4D instead of 2D)

### ISSUE THREAD

**Issue by mcupacTT** [tenstorrent/tt-metal#45962]
### Component / Area

ops

### Issue Type (optional)

None

### Observed

embedding_bw returns tensor<1x1xVxH> instead of tensor\<VxH\>.

When clip_grad_norm does sum(dim=[0,1], keep_dim=false), it reduces the
wrong dimensions (the leading 1x1 instead of VxH), producing a
tensor<VxH> instead of a scalar. 

The subsequent reshape tensor\<bf16\> → tensor<1xbf16>
then fails with: TT_FATAL: new_volume == old_volume

### Expected

embedding_bw should return tensor\<VxH\>

### 1. Steps (exact commands)

```
import torch
import ttnn

device = ttnn.open_device(device_id=0)

vocab_size = 1024
hidden_dim = 64
seq_len = 32

# embedding_bw requires grad to be 4D with first two dims = 1
grad_output_torch = torch.randn(1, 1, seq_len, hidden_dim, dtype=torch.bfloat16)
grad_output = ttnn.from_torch(grad_output_torch, device=device, layout=ttnn.TILE_LAYOUT)

input_ids_torch = torch.randint(0, vocab_size, (1, seq_len), dtype=torch.int32)
input_ids = ttnn.from_torch(input_ids_torch, dtype=ttnn.uint32, device=device, layout=ttnn.ROW_MAJOR_LAYOUT)

weight_torch = torch.randn(vocab_size, hidden_dim, dtype=torch.bfloat16)
weight = ttnn.from_torch(weight_torch, device=device, layout=ttnn.TILE_LAYOUT)

grad_weight = ttnn.embedding_bw(input_ids, weight, grad_output)

if isinstance(grad_weight, (list, tuple)):
    grad_weight = grad_weight[0]

print(f"=== embedding_bw output shape ===")
print(f"Expected shape: [{vocab_size}, {hidden_dim}]  (2D)")
print(f"Actual shape:   {grad_weight.shape}")

grad_sum = ttnn.sum(grad_weight, dim=[0, 1], keepdim=False)
print(f"\nAfter sum(dim=[0,1]): expected=scalar, actual shape={grad_sum.shape}")

ttnn.close_device(device)
```

Note: The full clip_grad_norm also does multiply (square) and sqrt, but those don't change the shape, therefore not being relevant for the repro.

### 2. Input data / link or description

None

### 3. Frequency

Always

### 1. Software Versions

tt-metal: commit c5ebc635

### 2. Hardware Details

Wormhole n150

### Is this a regression?

Unknown

### Logs & Diagnostics

Output of above repro:

```
=== embedding_bw output shape ===
Expected shape: [1024, 64]  (2D)
Actual shape:   Shape([1, 1, 1024, 64])

After sum(dim=[0,1]): expected=scalar, actual shape=Shape([1024, 64])
```

### Priority

P2

### Impact

Affects usage of clip_grad_norm in trainings
